### ISSUE TITLE
SCA 3D Secure

### ISSUE THREAD

**Issue by hugo49** [odoo/odoo#34253]
Impacted versions: 12

3D secure will be mandatory under a law of Europe soon for online payment. Is Odoo compatible? My credit card provider is ready. Modifications are to be expected in the configuration? 

"As of September 14th, the European regulations will require the 
strong client authentication 
 (SCA) for many online payments. When it comes into effect, a form of two-factor authentication will be required for a large number of card payments. If you do not update your integration, many transactions will be rejected by your customers' banks. "

**Comment by Feyensv** [comment]
It is in the task/objectives of Odoo R&D Sales Team ;), we aim to support it in time for V11-12-13 (and maybe also v10).

**Comment by Yenthe666** [comment]
As this has been answered I'm closing off the report. @Feyensv you're always welcome to link the commits here when it is in place 🎉

**Comment by yostashiro** [comment]
@Feyensv May we know what we should expect regarding this topic - the task scope in your team, the expected timeline to get solutions ready, and the actions that we are expected to take in existing Odoo installations?

My particular concern is what we should expect with `payment_stripe` which one of our customers uses in Europe.

**Comment by fgi-odoo** [comment]
We plan to release updates of our Stripe integration mid-August (for version 10 11 12). We switch to the new Checkout (form mode) and Intents API (s2s mode).

**Comment by fgi-odoo** [comment]
We will publish user guidelines to update the system.

**Comment by pedrobaeza** [comment]
And what about general payment workflow/other integrations?

**Comment by fgi-odoo** [comment]
No change is planned in the functional workflow. We are however also working on the technical side of authorize.net & ingenico integrations in order to be fully PCI compliant with the s2s mode.

**Comment by pedrobaeza** [comment]
OK, thanks. I meant other community integrations, like the one we have in Spanish localization (https://github.com/OCA/l10n-spain/tree/12.0/payment_redsys), as we don't know if general flows (technical ones) needs to be changed in order to accommodate this, and so that the module will need to be adapted as well.

Please give links when done for the rest for checking.

**Comment by Feyensv** [comment]
Some compliance changes will only modify targeted payment providers (ogone/stripe/...) but a generic task aims to review Odoo payment security and this one will break third party modules #34329.

**Comment by pedrobaeza** [comment]
@Feyensv thanks for the link. I'm seeing that is for master. Are there going to be changes in stable branches finally to accommodate SCA?

**Comment by Feyensv** [comment]
1. SCA/PCI changes will only change concerned payment acquirers (ogone, stripe, authorize, ...), the less possible, in v(10)-11-12-13.
It shouldn't have impacts on other modules than the concerned payment acquirers (and no change on the payment base module).

2. The PR of my last message does indeed target master because it a generic security check on the whole payment flow to verify the security, and apply a little bit of cleaning.
It is expected to break payment extensions from third-party modules but should only happen in master.

**Comment by pedrobaeza** [comment]
OK, thanks for the clarifications. Let me know if you need any review.

**Comment by Feyensv** [comment]
Last news about SCA in [Odoo Blog](https://www.odoo.com/blog/notre-blog-5/post/get-your-odoo-ready-for-strong-customer-authentication-sca-553)

**Comment by Yenthe666** [comment]
TLDR: 1 september 2019 it'll be released and a mailing will be done.

**Comment by pedrobaeza** [comment]
Thanks for the info!

**Comment by vinaysah** [comment]
Hello, Is this feature release for Odoo 12, If not then what is the current plan?
I am not able to find in 12.0 git branch.

**Comment by eLBati** [comment]
Hi @fgi-odoo any news about 10.0?

**Comment by fgi-odoo** [comment]
Will be ready in a few days. v11 and 12 are already OK with a new module to install.

**Comment by eLBati** [comment]
:eyes:

**Comment by atularvind** [comment]
👀

**Comment by eLBati** [comment]
@atularvind https://github.com/odoo/odoo/tree/12.0/addons/payment_stripe_sca
